### ISSUE TITLE
Implement selectionContainer design pattern for lists and tree views

### ISSUE THREAD

**Issue by leonardder** [nvaccess/nvda#10861]
### Is your feature request related to a problem? Please describe.
The people from VS Code are struggling with how to communicate selection in their multiselectable tree view to NVDA. Currently, only unselected items are mentioned as such, multi selection isn't reported. See https://github.com/microsoft/vscode/issues/91061

### Describe the solution you'd like
I'd like to expand the selectionContainer approach to lists and tree views, in order for:

* Not selected to be spoken for unselected items
* Selected to be spoken when more than one item is selected
* Multi selection to be pronounced as such, e.g. by means of a new state.

Thoughts @michaelDCurran and @jcsteh?

Cc @isidorn

**Comment by feerrenrut** [comment]
Could you expand the description? Do you intend to change behavior only for vsCode, or more generally? I quickly read through microsoft/vscode#91061 and it seems you already noted that we don't speak "not selected" for verbosity reasons, since it is the default state for most lists. Why is it necessary to have a different approach in vsCode? Could you summarize the rationale in the linked issue for us?

**Comment by leonardder** [comment]
I updated the initial description.

> Why is it necessary to have a different approach in vsCode? Could you summarize the rationale in the linked issue for us?

Sure. The explorer tree view in Visual Studio Code allows multiple items to be selected. Currently, the selected state is omitted for all selected items, whether selected one or more items. Announcing the selected state when multiple items in the tree are selected (such as in Google Sheets) would be helpful, not only for VS Code but also for other lists that allow selection.

**Comment by leonardder** [comment]
Ugh, it looks like VS code selection really behaves a bit weird from a non visual perspective, see https://github.com/microsoft/vscode/issues/91061#issuecomment-618883146 and https://github.com/microsoft/vscode/issues/91061#issuecomment-618900791.

@bkoray noted in https://github.com/microsoft/vscode/issues/91061#issuecomment-618624912 that he would really like the ability in NVDA to hear how many items are selected. Though that's a different subject, I think this issue can be a good starting point. As soon as the list/tree views in VS Code behave like selection containers, we can expand the report object with focus script to provide a selection count, for example.

**Comment by lukaszgo1** [comment]
Its not only problem in VS Code tree views in JetBrains IDE's are suffering from similar issues. There the  experience is worse as every item with focus is announced as selected, unless unselected explicitly with CTRL+Space

**Comment by Adriani90** [comment]
Related to #14352. Chrome started to improve things, at least for single select.
cc: @SaschaCowley
